### PR TITLE
Add CIRCLE and fixup Linux tests

### PR DESCRIPTION
**Pull request by MrLotU**

### CODE REVIEW

**Comment by calebkleveter** [comment]
@tanner0101 Would you mind enabling Circle for this repo?

File: Tests/LinuxMain.swift [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import XCTest
 
-import nio_kitTests
+import NIOKit
 
 var tests = [XCTestCaseEntry]()
-tests += nio_kitTests.allTests()
+tests += NIOKit.allTests()
```
Comment (calebkleveter) [inline Tests/LinuxMain.swift]: This should be `NIOKitTests`

Comment (MrLotU) [reply Tests/LinuxMain.swift]: Fixed :)

**Comment by MrLotU** [comment]
Sounds good 👍

File: circle.yml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      - run: 
+          name: Compile code
+          command: swift build
+      - run: 
+          name: Run unit tests
+          command: swift test
+
+  linux-41:
+    docker:
+      - image: codevapor/swift:4.1
```
Comment (tanner0101) [inline circle.yml]: We should use the new `5.0` image :)

File: circle.yml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+workflows:
+  version: 2
+  tests:
+    jobs:
+      - linux
+      - linux-41
+      - linux-release
+       - macos
+
+  nightly:
```
Comment (tanner0101) [inline circle.yml]: I don't think we need nightly really going forward. Haven't gotten much use out of them.

File: circle.yml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+      - run: swift test
+
+  linux:
+    docker:
+      - image: swift:4.2
+    steps:
+      - checkout
+      - run: 
+          name: Compile code
+          command: swift build
```
Comment (tanner0101) [inline circle.yml]: we're probably ok to remove the `name:, command:` labels here and match the `macOS` section. Would be easier to read I think.

**Comment by tanner0101** [comment]
https://github.com/vapor/vapor/blob/master/Tests/LinuxMain.swift
https://github.com/vapor/vapor/blob/master/Tests/VaporTests/ApplicationTests.swift#L742-L745

Here are relevant references for setting up the testCase stuff

**Comment by MrLotU** [comment]
@tanner0101 @calebkleveter Could you both have another look? CI works now :D

**Pull request merged**